topdown/cidr.go

```diff
@@ -332,25 +332,26 @@ func evalNetCIDRMerge(networks []*net.IPNet) []*net.IPNet {
 }
 
 func generateIPNet(term *ast.Term) (*net.IPNet, error) {
-	switch e := term.Value.(type) {
-	case ast.String:
-		network := &net.IPNet{}
-		// try to parse element as an IP first, fall back to CIDR
-		ip := net.ParseIP(string(e))
-		if ip != nil {
-			network.IP = ip
-			network.Mask = ip.DefaultMask()
-		} else {
-			var err error
-			_, network, err = net.ParseCIDR(string(e))
-			if err != nil {
-				return nil, err
-			}
-		}
-		return network, nil
-	default:
+	e, ok := term.Value.(ast.String)
+	if !ok {
 		return nil, errors.New("element must be string")
 	}
+
+	// try to parse element as an IP first, fall back to CIDR
+	ip := net.ParseIP(string(e))
+	if ip == nil {
+		_, network, err := net.ParseCIDR(string(e))
+		return network, err
+	}
+
+	network := &net.IPNet{IP: ip}
+	switch len(ip) {
+	case net.IPv4len:
+		network.Mask = ip.DefaultMask()
+	case net.IPv6len:
+		network.Mask = net.CIDRMask(128, 128) // "/128"
```

Comment:
If only IPv4 addresses have default masks, should we assume this or would it be better to return an error if one is not provided ?

Reply:
Yeah good point. I'll change this to return a (non-halt) error if the IPv6 input doesn't come with a prefix and can't be mapped to an IPv4 (which would have a default mask).